packages/ember-data/lib/initializers/store.js

```diff
@@ -12,13 +12,20 @@ import Store from "ember-data/system/store";
   @param {Object} [application] an application namespace
 */
 export default function initializeStore(registry, application) {
+  // initializerStore be called by an Ember.initializer
+  // when used with ember 1.11.0 or below.
+  var params = getRegistryAndContainer(registry, application);
+  application = params.application;
+  registry = params.registry;
+  var container = params.container;
+
   Ember.deprecate('Specifying a custom Store for Ember Data on your global namespace as `App.Store` ' +
                   'has been deprecated. Please use `App.ApplicationStore` instead.', !(application && application.Store));
 
   registry.optionsForType('serializer', { singleton: false });
   registry.optionsForType('adapter', { singleton: false });
 
-  registry.register('store:main', registry.lookupFactory('store:application') || (application && application.Store) || Store);
+  registry.register('store:main', container.lookupFactory('store:application') || (application && application.Store) || Store);
```

Comment:
I think you are right and `application.register` is correct in the new model. However, that will break when ED is used with older version of Ember. I'll add a comment to the codebase to take this out when we drop support for old Ember.